rpc-protocol/src/main/java/org/sonarsource/sonarlint/core/rpc/protocol/SonarLintLauncherBuilder.java

```diff
@@ -49,10 +55,9 @@ protected MessageJsonHandler createJsonHandler() {
       public GsonBuilder getDefaultGsonBuilder() {
         // We don't want the EnumTypeAdapter from lsp4j, as we want to serialize enums as string (this is the default in Gson)
         return new GsonBuilder()
-          .registerTypeAdapterFactory(new CollectionTypeAdapter.Factory())
-          .registerTypeAdapterFactory(new ThrowableTypeAdapter.Factory())
           .registerTypeAdapterFactory(new EitherTypeAdapter.Factory())
-          .registerTypeAdapterFactory(new TupleTypeAdapters.TwoTypeAdapterFactory())
+          .registerTypeAdapterFactory(new EitherServerOrLocalHotspotAdapterFactory())
```

Comment:
The order is important here: the last added will have higher priority. So we should add more specific Either adapters after the generic `EitherTypeAdapter`.

Reply:
Maybe worth adding a comment in the code?